pvlib/data/gold/sdm.py

```diff
@@ -0,0 +1,598 @@
+    """
+    Create a x-Si device based on cell parameters in Bishop's 1988 paper Fig. 6.
+
+    Parameters are normalized to the cell area.
+    """
+
+    # Copy an existing mono-Si device and None the entries we don't know or need
+    device = copy.deepcopy(pvsystem.retrieve_sam(
+        'cecmod').SunPower_SPR_E20_327)
+    device.name = 'Bishop_{}_{}_{}'.format(N_s, cell_area_cm2, ideality_factor)
+    device.BIPV = 'N'
+    device.Date = '6/20/1988'
+    device.T_NOCT = None
+    device.A_c = None
+    device.N_s = N_s
+    device.I_sc_ref = None
+    device.V_oc_ref = None
+    device.I_mp_ref = None
+    device.V_mp_ref = None
+    device.beta_oc = None
+    device.a_ref = N_s * ideality_factor * constants.boltzmann_J_per_K * \
+        constants.T_stc_K / constants.elementary_charge_C  # Volt
+    device.I_L_ref = cell_area_cm2 * 30.e-3  # Amp
+    device.I_o_ref = cell_area_cm2 * 5500.e-12  # Amp
+    device.R_s = N_s * cell_area_cm2 * 1.33  # Ohm
+    device.R_sh_ref = cell_area_cm2 * 750. / N_s  # Ohm
+    device.Adjust = None
+    device.gamma_r = None
+    device.Version = None
+    device.PTC = None
+
+    return device
+
+
+def bootstrap_si_device_pvmismatch(N_s=1, cell_area_cm2=153., ideality_factor=1.5):
+    """
+    Create a x-Si device based on default cell parameters in PVMismatch.
+    
+    Parameters are not normalized to the cell area.
+    First diode taken from two diode model used in PVMismatch, and default ideality factor
+      is average of first and second diode ideality factors.
+    https://github.com/SunPower/PVMismatch/blob/master/pvmismatch/pvmismatch_lib/pvcell.py
+    """
+
+    # This should match the default value for cell_area_cm2 argument
+    default_cell_area_cm2 = 153.
+
+    # Copy an existing mono-Si device and None the entries we don't know or need
+    device = copy.deepcopy(pvsystem.retrieve_sam(
+        'cecmod').SunPower_SPR_E20_327)
+    device.name = 'PVMismatch_{}_{}_{}'.format(
+        N_s, cell_area_cm2, ideality_factor)
+    device.BIPV = 'N'
+    device.Date = '2/20/2018'
+    device.T_NOCT = None
+    device.A_c = None
+    device.N_s = N_s
+    device.I_sc_ref = None
+    device.V_oc_ref = None
+    device.I_mp_ref = None
+    device.V_mp_ref = None
+    device.beta_oc = None
+    device.a_ref = N_s * ideality_factor * constants.boltzmann_J_per_K * \
+        constants.T_stc_K / constants.elementary_charge_C  # Volt
+    device.I_L_ref = cell_area_cm2 * 6.3056/default_cell_area_cm2  # Amp
+    device.I_o_ref = cell_area_cm2 * 2.286188161253440e-11/default_cell_area_cm2  # Amp
+    device.R_s = N_s * cell_area_cm2 * \
+        0.004267236774264931/default_cell_area_cm2  # Ohm
+    device.R_sh_ref = cell_area_cm2 * 10.01226369025448 / \
+        default_cell_area_cm2 / N_s  # Ohm
+    device.Adjust = None
+    device.gamma_r = None
+    device.Version = None
+    device.PTC = None
+
+    return device
+
+
+def sdm_make_gold_dataset(json_filepath=None):
```

Comment:
recommend removing the option to write the file from within this function. I think it's better for this function to do it's one thing (return the data) and for another function to write (and maybe verify) the file.